### DISCUSSION TITLE
Send commands to specific existing pane

### DISCUSSION THREAD

**Discussion by jamesgecko** [zellij-org/zellij#2704] [Q&A]
I have a Tmux setup where I have vim in one pane and bash in another. When I press a key combination, vim sends a command to run the current spec to the bash pane.

Is there a way to do this in Zellij? `zellij run` seems to only create new panes, and `zellij action` can find tabs by name, but not panes by name. Is something like this my best option, to just jump to whatever the next pane is to write the command? `zellij action focus-next-pane && zellij action write-chars "bundle exec rspec path/to/my/file" && zellij action write 13 && zellij action focus-previous-pane`?

**Comment by zenom** [comment]
Did you find a solution to this? Working down this path myself and running into similar issues.

**Comment by nerdictranslator** [comment]
Same deal. Looking for a way to do this. Ah well. I guess it's a dream for another day.

**Comment by vimkim** [comment]
This is one of the features I want the most.

**Comment by lassoColombo** [comment]
Related: i'm looking for a way to focus a pane by name.

My usecase is this: I have configured a keybinding to spawn a floating pane opening lazygit. If i press the same binding a second time it will spawn a second instance of the program, which is undesired. 
Ideally l'd like to be able to check if the lazygit pane exists in order to either create it or focus it

Any update on this?

**Comment by quantonganh** [comment]
I've just found these (but haven't tried them yet):
- https://zellij.dev/documentation/plugin-api-commands.html#write_chars_to_pane_id
- https://zenn.dev/atani/articles/zellij-send-keys-tmux-like-plugin

**Comment by mrshu** [comment]
I've put [`zjctl`](https://github.com/mrshu/zjctl) together exactly because I was missing something like this. Try it out and let me know if it helps / how it can be improved!

**Comment by atani** [comment]
Author of [zellij-send-keys](https://github.com/atani/zellij-send-keys) here.

This plugin was built exactly for this use case — sending text/commands to a specific pane by ID, like `tmux send-keys`.

```bash
# Quick example
source scripts/setup-env.sh
send-to-pane 0 "bundle exec rspec spec/models/user_spec.rb"
```

It works via `zellij action pipe`, so you can call it from outside a session too (using `ZELLIJ_SESSION_NAME`). Supports both terminal and plugin panes.

Would love feedback from anyone trying it out.
